src/components/action-popover/action-popover-item/action-popover-item.component.tsx

```diff
@@ -139,19 +143,106 @@ export const ActionPopoverItem = ({
   const mouseEnterTimer = useRef<NodeJS.Timeout | null>(null);
   const mouseLeaveTimer = useRef<NodeJS.Timeout | null>(null);
 
+  const currentSubmenuAlignment = () => {
+    if (submenuPosition === "left") {
+      if (leftAlignmentState) {
+        return "right";
+      }
+      return "left";
+    }
+    return "right";
+  };
+
+  function getContainerPosition(
+    itemRef: React.RefObject<HTMLElement>,
+    containerSubmenuRef: React.RefObject<HTMLElement>,
+    containerPlacement: "bottom" | "top"
+  ): ContainerPosition | undefined {
+    /* istanbul ignore if */
+    if (!itemRef.current || !containerSubmenuRef.current) return undefined;
+
+    const { offsetWidth: submenuWidth } = containerSubmenuRef.current;
+
+    const horizontalAlignmentValue = currentSubmenuAlignment() === "left";
+    const leftValue = horizontalAlignmentValue ? -submenuWidth : "auto";
+    const rightValue = horizontalAlignmentValue ? "auto" : -submenuWidth;
+    const yPositionName = containerPlacement === "top" ? "bottom" : "top";
+
+    return {
+      left: leftValue,
+      [yPositionName]: "calc(-1 * var(--spacing100))",
+      right: rightValue,
+    };
+  }
+
+  const retrieveVariantType = () => {
+    if (!iconState && !caretState && !icon && !submenu) {
+      return 1;
+    }
+    if (!iconState && caretState && !icon && !submenu) {
+      return 2;
+    }
+    if (!iconState && caretState && !icon && submenu) {
+      return 3;
+    }
+    if (iconState && !caretState && !icon && !submenu) {
+      return 4;
+    }
+    if (iconState && !caretState && icon && !submenu) {
+      return 5;
+    }
+    if (iconState && caretState && icon && submenu) {
+      return 6;
+    }
+    if (iconState && caretState && icon && !submenu) {
+      return 7;
+    }
+    if (iconState && caretState && !icon && submenu) {
+      return 8;
+    }
+    if (iconState && caretState && !icon && !submenu) {
+      return 9;
+    }
+    return 1;
+  };
+
   useEffect(() => {
     if (!isOpenPopover) {
       setOpen(false);
     }
   }, [isOpenPopover]);
 
+  useEffect(() => {
+    if (icon) {
+      setIconState?.(true);
+    }
+    if (submenu) {
+      setCaretState?.(true);
+    }
+  }, [icon, setCaretState, setIconState, submenu]);
+
+  useEffect(() => {
+    if (!isLeftAligned) {
+      setLeftAlignmentState?.(true);
+    }
+
+    if (isLeftAligned) {
```

Comment:
**suggestion**: if we do need to keep this `useEffect`, this can just be an `if/else` rathre than 2 `if`s - or even just a single line: `setLeftAlignmentState(!isLeftAligned)`.

(Also that makes me reaise something is backwards between these 2 booleans - seems weird, not sure what's going on there...)

Reply:
I'll have a look to see if this can be changed to an `else` 👍